src/main/kotlin/io/fuchs/gradle/collisiondetector/CollisionDetectorPlugin.kt

```diff
@@ -5,17 +5,18 @@ import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.util.GradleVersion
 
-open class CollisionDetectorPlugin : Plugin<Project> {
+abstract class CollisionDetectorPlugin : Plugin<Project> {
 
-    val taskName: String = "detectCollisions"
+    private val taskName: String = "detectCollisions"
 
     override fun apply(project: Project) {
-        if (GradleVersion.current() < GradleVersion.version("4.0")) {
-            throw GradleException("This plugin only supports Gradle 4.0 or newer versions")
+        if (GradleVersion.current() < GradleVersion.version("6.6")) {
+            throw GradleException("This plugin only supports Gradle 6.6 or newer versions")
         }
 
-        project.pluginManager.withPlugin("java") {
-            project.tasks.create(taskName, DetectCollisionsTask::class.java)
+        project.tasks.register(taskName, DetectCollisionsTask::class.java) {
+            // convention useful for most Jars, can be replaced completely using 'setExcludes()'
+            it.collisionFilter.exclude("META-INF/**", "module-info.class")
```

Comment:
I have used this plugin the past in combination with the [Gradle Shadow](https://github.com/johnrengelman/shadow) plugin, which merges many JARs into a single one, as I have run into issues when Service Descriptor files under `META-INF/services` were colliding and overwriting each other.

There is an [option in that plugin ](https://imperceptiblethoughts.com/shadow/configuration/merging/#merging-service-descriptor-files) to enable merging of such files, which is helpful but does not happen by default.

This change would mean that we do not detect that situation (collision of these service descriptors) anymore by default. Can we easily include `META-INF/services/**` but exclude everything else in `META-INF`? In any case, I believe the convention here is useful.

Reply:
Good point. I did not think about such a use case.

> an we easily include META-INF/services/** but exclude everything else in META-INF?

This would be possible, but I think it would be an inconsistent default for the (standard?) use case where you check a "normal" classpath. Because then the duplicated service loader descriptors are allowed, expected, and used. Similar to how multiple `module-info.class` or `MANIFEST.MF` files are used.

In my opinion we should then either:

1. Use the default as suggested in this PR, which fits the use case of _"I apply the plugin and give it my runtime classpath and expect it **not** to fail if there are **no** problematic duplicates"_
2. Set no default at all, so that you always have to configure the plugin (depending on what you want to use it for)

I would still tend towards option 1, but we could also do 2.

Reply:
Another thing about conventions: 
I used the plugin a bit more, thinking about how it is most easy to explain what it does.

In that context, I got the idea that it would be a good default to set it to analyze the _runtimeClasspath_ when used in a Java project.

Then you can just apply the plugin, run `:detectCollisions` and immediately get results without the need to configure anything. I think this makes it very accessible.

But let me know if you don't like that, then I can remove it from the PR again.

Reply:
I agree with both your points on the conventions.